### PR TITLE
Add generated documentation in markdown file

### PR DESCRIPTION
**Pull request by zuiderkwast**
Fixes broken 'rebar3 edoc' and add generated files to repo. Helper
modules are marked as private, to prevent them from being included
in the docs.

----

Using `edown`, the EDoc (`@doc`) tags in the source code is used to generate markdown files under doc/. I've hidden all modules except the public API module.

Activating this is the first step. Then we can improve the documentation itself.

Review the generated documentation: https://github.com/zuiderkwast/eredis_cluster/blob/markdown-manual/doc/eredis_cluster.md

I think we should move some of the contents in the README to other markdown files under doc/ as well.

### CODE REVIEW

File: doc/eredis_cluster.md [hunk 1]
```diff
@@ -0,0 +1,466 @@
+
```
Comment (bjosv) [inline doc/eredis_cluster.md:1]: a bit weird that there is no comment that this file shouldn't be changed manually..but I guess thats how edoc is

File: Makefile [hunk 2]
```diff
@@ -48,14 +48,15 @@ coverview: cover
...
 edoc:
-	@$(REBAR) skip_deps=true doc
+	@$(REBAR) edoc
 
 help:
 	@echo "Please use 'make <target>' where <target> is one of"
 	@echo "  start             starts a test redis cluster"
 	@echo "  stop              stops all redis servers"
 	@echo "  travis-run        starts the redis cluster and runs your tests"
 	@echo "  travis-install    install redis from 'unstable' branch"
```
Comment (bjosv) [inline Makefile:58]: Seems that the target travis-run dont even exists

**Review by bjosv — APPROVED** [review]
LGTM, but I guess you are updating the edoc tags

File: doc/eredis_cluster.md (hunk 1 above)
Comment (zuiderkwast) [reply doc/eredis_cluster.md:1]: EDoc normally generates HTML. https://github.com/uwiger/edown is used for generating Markdown and it's not perfect.

File: Makefile (hunk 2 above)
Comment (zuiderkwast) [reply Makefile:58]: travis-run exists but not travis-install. Let's update (or delete) those when we get travis or github actions running.

**Pull request merged**